src/hooks/useConnectOwnedVenues.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { createContext, useContext, useMemo, useCallback } from "react";
+
+import { SparkleSelector } from "types/SparkleSelector";
+import { ReactHook } from "types/utility";
+import { AnyVenue } from "types/venues";
+
+import { WithId, withId } from "utils/id";
+
+import { useFirestoreConnect, isLoaded } from "hooks/useFirestoreConnect";
+import { useSelector } from "hooks/useSelector";
+
+export const ownedVenuesSelector: SparkleSelector<
+  Record<string, AnyVenue> | undefined
+> = (state) => state.firestore.data.ownedVenues;
+
+export const useConnectOwnedVenues = (userId?: string) => {
+  useFirestoreConnect(
+    userId
+      ? {
+          collection: "venues",
+          where: [["owners", "array-contains", userId]],
+          storeAs: "ownedVenues",
+        }
+      : undefined
+  );
+};
+
+export interface OwnedVenuesContextState {
+  isLoaded: boolean;
+  isLoading: boolean;
+
+  ownedVenues: WithId<AnyVenue>[];
+  ownedVenuesIds: string[];
+
+  findVenueInOwnedVenues: (
+    searchedForVenueId: string
+  ) => WithId<AnyVenue> | undefined;
+}
+
+const OwnedVenuesContext = createContext<OwnedVenuesContextState | undefined>(
+  undefined
+);
+
+export interface OwnedVenuesProviderProps {
+  userId?: string;
+}
+
+export const OwnedVenuesProvider: React.FC<OwnedVenuesProviderProps> = ({
+  children,
+  userId,
+}) => {
+  useConnectOwnedVenues(userId);
+
+  const venues = useSelector(ownedVenuesSelector);
+
+  const ownedVenues = useMemo(
+    () => Object.entries(venues ?? {}).map(([id, venue]) => withId(venue, id)),
+    [venues]
+  );
+
+  const ownedVenuesIds = useMemo(() => Object.keys(venues ?? {}), [venues]);
+
+  const findVenueInOwnedVenues = useCallback(
+    (searchedForVenueId: string): WithId<AnyVenue> | undefined =>
+      ownedVenues.find((venue) => venue.id === searchedForVenueId),
+    [ownedVenues]
```

Comment:
If we are using useFirestoreConnect, there is no need in Context/Provider. This was done for users only because it was the first step to improve performance. After that the useConnect was removed from there

Reply:
This should be done similar to `auditorium/useAllAuditoriumSections`

Reply:
removed the Context/Provider 8819037c8d4dc531a31279aa15a3e1979048a188